ui/app/AppLayouts/Wallet/WalletLayout.qml

```diff
@@ -139,19 +137,26 @@ Item {
         function displayAllAddresses() {
             RootStore.showSavedAddresses = false
             RootStore.selectedAddress = ""
-            RootStore.setFillterAllAddresses()
+            RootStore.setFilterAllAddresses()
         }
 
         function displayAddress(address) {
             RootStore.showSavedAddresses = false
             RootStore.selectedAddress = address
+            d.resetRightPanelStackView() // Avoids crashing on asset items being destroyed while in signal handler
```

Comment:
Could you please share more information what is this about? It sounds that we hide an issue and fix a side effect here.

Reply:
> the reason was that a chain of signal handlers happen when setFilterAddress is called below

I don't like at all the way how we're refreshing the wallet, there are too many dependencies that are triggered on refresh. I had a similar issue when wanted to implement a log-out feature.

Reply:
> ...
> So to me the fix makes sense, but I believe it could be fixed in some other way to stop chained signals at some point, but it will require more time, so if you think this solution is too hacky, we might want to create an issue for that. WDYT?

Adding an issue with descriptions of what you learned to follow up on the root cause of this crash sounds like a good way to stay safe in the long run.